social_django/migrations/0017_usersocialauth_user_social_auth_uid_required.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 5.1.7 on 2025-03-14 12:16
+
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("social_django", "0016_alter_usersocialauth_extra_data"),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.AddConstraint(
+            model_name="usersocialauth",
+            constraint=models.CheckConstraint(
+                check=models.Q(("uid", ""), _negated=True),
+                name="user_social_auth_uid_required",
+            ),
```

Comment:
The same `django.VERSION` approach can be used here

```suggestion
            constraint=(
                models.CheckConstraint(condition=~models.Q(uid=""), name="user_social_auth_uid_required")
                if django.VERSION >= (5, 1)
                else models.CheckConstraint(check=~models.Q(uid=""), name="user_social_auth_uid_required")
            ),
```